controllers/chat/sendSignedMessage.js

```diff
@@ -84,6 +86,25 @@ const sendSignedMesage = async (req, res) => {
     if (createdChannel?.channel?.is_channel_blocked)
       return res.status(403).json(responseError('Channel is blocked'));
 
+    // Prevent if user try to chat with blocked user
+    let channelType = createdChannel?.channel?.channel_type;
+    if (channelType === 4 || channelType === 0) {
+      const blockedIds = await UsersFunction.getBlockedAndBlockerUserId(
+        UserBlockedUser,
+        req.userId
+      );
+      let better_channel_member = createdChannel?.channel?.better_channel_member;
+      if (blockedIds.length > 0) {
+        for (const member of better_channel_member) {
+          if (member.user_id !== req.userId && blockedIds.includes(member.user_id)) {
+            return res
+              .status(403)
+              .json(responseError('This user does not want to receive messages'));
+          }
+        }
+      }
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for checking if a user is blocked seems to be correct, but it could be optimized. Currently, you are looping through all the members of `better_channel_member` and checking if each member is in the `blockedIds` list. This results in a time complexity of O(n^2) which can be quite inefficient when there are many members or blocked users.

A more efficient approach would be to convert `blockedIds` into a Set, which allows for constant time lookup, and then loop through the `better_channel_member` array. This reduces the time complexity to O(n).

Here's how you can implement this:

```diff
-       let better_channel_member = createdChannel?.channel?.better_channel_member;
-       if (blockedIds.length > 0) {
-         for (const member of better_channel_member) {
-           if (member.user_id !== req.userId && blockedIds.includes(member.user_id)) {
-             return res
-               .status(403)
-               .json(responseError('This user does not want to receive messages'));
-           }
-         }
-       }

+       const blockedIdsSet = new Set(blockedIds);
+       const better_channel_member = createdChannel?.channel?.better_channel_member;
+       for (const member of better_channel_member) {
+         if (member.user_id !== req.userId && blockedIdsSet.has(member.user_id)) {
+           return res
+             .status(403)
+             .json(responseError('This user does not want to receive messages'));
+         }
+       }
```
Please note that this change will improve the performance of your code without altering its functionality.


<!-- This is an auto-generated comment by OSS CodeRabbit -->